sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V1FallbackWriters.scala

```diff
@@ -38,10 +38,11 @@ case class AppendDataExecV1(
     table: SupportsWrite,
     writeOptions: CaseInsensitiveStringMap,
     plan: LogicalPlan,
-    refreshCache: () => Unit) extends V1FallbackWriters {
+    refreshCache: () => Unit,
+    override val write: Option[V1Write] = None) extends V1FallbackWriters {
 
-  override protected def run(): Seq[InternalRow] = {
-    writeWithV1(newWriteBuilder().buildForV1Write(), refreshCache = refreshCache)
+  override protected def buildAndRun(): Seq[InternalRow] = {
```

Comment:
We can get rid of all `buildAndRun` methods if we are OK to apply the new logic all the time.

Reply:
I'm interested to hear what @dongjoon-hyun thinks about this.

I think we should have a different physical node for each write so that the explain plan shows what is happening. Otherwise, the approach to support building the batch write here or building it in the optimizer was mainly to be able to turn this on and off in our environment. I doubt that is needed in other situations.

I think I would be for removing all of the `buildAndRun` methods and always building the write in the optimizer.

Reply:
Does it cause many code change on top of this? If it is not intrusive, it sounds reasonable in that context. I'd give +1 for the direction, @rdblue .

Reply:
Getting rid of `buildAndRun` would also ensure we don't have to maintain the same logic in two places.

Reply:
I'm also +1 on getting rid of `buildAndRun`.

Reply:
It sounds like an intermediate consensus, I'll update the PR but we can revisit this once we have more input from others.